maturity-model.md

```diff
@@ -0,0 +1,175 @@
+Leaders use those channels themselves, openly encourage others to use them, and maintain team-facing or public-facing records of the feedback they've received and/or the actions they've taken to address this feedback.
+
+**Leadership**
+
+* LS-0:	Command & control culture, within a highly hierarchical organization.
+* LS-1: Some leaders are open to receiving feedback and creating an environment where people feel safe providing it.
+* LS-2:	Most leaders in the organization are open to receiving feedback and creating an environment where people feel safe providing it. Leaders show passivity about understanding whether all members feel empowered and enabled to share. Organization encourages leaders to actively seek voices not present in dialog out for inclusion.
+* LS-3: Members feel empowered and enabled to share opinions constructively on any matter relevant to their work or about which they feel passionate.
+
+**Organizational and Functional Structure**
+
+* OF-0:	Working groups tend to be static in terms of membership and skill sets.
+* OF-1:	Cross-functional teams exist, but team roles are often unclear and governance structures are vague.
+* OF-2:	Cross-functional teams are common, and teams post their roles and goals publicly.
+* OF-3:	Cross-functional teams are common and make their activities known broadly to the organization; in turn, the organization promotes best practices for working together.
+
+**Contribution**
+
+* CB-0:	Completly siloed, no collaboration outside the teams. Just some collaborations due to cross-functional teams.
+* CB-1:	Members of the organization and teams collaborate but frequently say it's "too difficult". Teams infrequently revisit the outcomes of their collaborations.
+* CB-2:	Members of the organization and teams actively seek opportunities to collaborate. Teams routinely discuss, revisit and debate the outcomes of their collaborative efforts, and make these outcomes available by default.
+* CB-3:	Members of the organization collaborate both internally and externally in ways that benefit all involved. Teams routinely discuss, revisit and debate the outcomes of their collaborative efforts, and share their learnings outside the organization, and make these outcomes externally available by default.
+
+### Community
+
+**Sharing Policies**
+
+* SP-0: No sharing culture nor written policies.
+* SP-1:	Some members of the organization unite to define values and principles, but are not clearly supported when they do.
+* SP-2:	Members of the organization collectively document shared visions and agreements like mission statements and codes of conduct, make them easily accessible, and reference them often. Onboarding materials and orientation rituals provide adequate context for helping new members understand how the organization will benefit from their contributions.
```

Comment:
This sounds more like "how mature is the entire organisation to adopt InnerSource" than "how mature is the InnerSource project itself". The deeper I dive, I think these two questions should be separated. In some of the points above we seem to be speaking about the setup of individual InnerSource projects. In others about the entire organisation.

Reply:
The goal here is to have a common way of behaving across the organization. And for this, there is a need to build certain set of rules and guidelines that everyone agrees, knows and is able to reference if needed at some point.

When discussing about InnerSource in the previous sections, the goal at the very end is to have a common set of rules and guidelines we can all discuss about, share and reference if needed. It happens that depending on the level, it may make sense to start at the project level or at the organization level.

I see for instance that if we have a new tool, let's say Slack/Mattermost/RocketChat/etc.. to communicate internally, this should/may be fostered by the organization or the InnerSource program within the organization. And this is another tool that will hopefully make more productive/collaborative/communicative the several developers and contributors.

On the other hand, there might be cases where at the level of project, a specific group of people or the trusted committer decides that when coming to my project in my internal infrastructure, you should behave in a certain way and that the code review process takes place in a certain way due to project constraints. These constraints are project related and not organization related, while certain behaviors make more sense to have them discussed at the organization level.

Summarizing: the approach is top down or down top depending on the case and the maturity level. We may specify this specific context in each of the maturity levels if you think this is useful.

An example of this would be to state (just an example) that communication is a more top down approach where the InnerSource initiative will provide the tooling, training, and certain guidelines (once the community start using that tool). And then the specific guidelines about how to contribute to a certain project, although narrowed by the specific infra used (Gerrit, PRs, MRs, etc), are decided by each of the InnerSource projects (e.g., having 2 +2 positive votes as done in OpenStack instead of only one +2 vote).

What do you think about this?

Reply:
I think I understand now better where you are going with this point. I've tried to capture it in the motivation for that point in the model. Feedback welcome.

Reply:
And thank you for taking the time to type up the explanation above.